cmake/horovod/CMakeLists.txt

```diff
@@ -46,11 +40,30 @@ file(GLOB_RECURSE horovod_common_src
 
 add_library(horovod ${horovod_common_src})
 
-target_compile_options(horovod PRIVATE ${MPI_CXX_COMPILE_FLAGS})
-target_compile_options(horovod PRIVATE -march=native)
-target_compile_options(horovod PRIVATE -Wno-error=sign-compare)
-target_compile_options(horovod PRIVATE -Wno-reorder)
-target_compile_options(horovod PRIVATE -Wno-unused-variable)
+target_compile_definitions(horovod PRIVATE
+    EIGEN_MPL2_ONLY=1
+    HAVE_CUDA=1
+    HAVE_NCCL=1
+    HAVE_MPI
+    HOROVOD_GPU_ALLREDUCE='N'
+    HOROVOD_GPU_ALLGATHER='M'
+    HOROVOD_GPU_BROADCAST='M')
+
+target_compile_options(horovod PRIVATE
+    -pthread
```

Comment:
Nit: You can keep "-pthread" there but it isn't useful on Linux distros with Glibc.  Glibc doesn't need it starting from late 1990s.